### ISSUE TITLE
Using this in an arrow function in a class

### ISSUE THREAD

**Issue by jay1337** [eslint/eslint#9766]
**Tell us about your environment**

* **ESLint Version:** 4.13.1
* **Node Version:** 8.9.1
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
	extends: [
		'eslint:all',
		'plugin:import/errors',
		'plugin:import/warnings',
	],
	parserOptions: {
		ecmaVersion: 6,
		sourceType: 'module',
		allowImportExportEverywhere: true,
	},
	env: {
		es6: true,
	},
	parser: 'babel-eslint',
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
class SomeClass {
  anArrowFunctionWithThis = () => this.something = "some text";
}
```

```bash
node eslint.js js_file.js
```

**What did you expect to happen?**
No error from `no-invalid-this` rule. This pattern is quite useful and used with React : https://reactarmory.com/answers/when-to-use-arrow-functions

**What actually happened? Please include the actual, raw output from ESLint.**
There is an error: `Unexpected 'this'. (no-invalid-this)`

**Comment by not-an-aardvark** [comment]
Closing as a duplicate of https://github.com/eslint/eslint/issues/8271

**Comment by jay1337** [comment]
Thanks @not-an-aardvark ! You should add a link to https://github.com/babel/eslint-plugin-babel in the doc !
